protocols/rendezvous/examples/register.rs

```diff
@@ -39,14 +42,22 @@ async fn main() {
 
     let identity = identity::Keypair::generate_ed25519();
 
-    let mut swarm = Swarm::new(
-        development_transport(identity.clone()).await.unwrap(),
+    let mut swarm = SwarmBuilder::new(
+        tcp::tokio::Transport::default()
+            .upgrade(Version::V1)
+            .authenticate(noise::NoiseAuthenticated::xx(&identity).unwrap())
+            .multiplex(yamux::YamuxConfig::default())
+            .boxed(),
         MyBehaviour {
             rendezvous: rendezvous::client::Behaviour::new(identity.clone()),
             ping: ping::Behaviour::new(ping::Config::new().with_interval(Duration::from_secs(1))),
         },
         PeerId::from(identity.public()),
-    );
+    )
+    .executor(Box::new(|f| {
```

Comment:
Curious Thomas, why do we need it here but not when using `tcp::async_io::Tcp` ?

Reply:
`async-std` implicitly creates a reactor on each thread that an IO component is polled on. tokio only creates a reactor on worker threads and thus requires explicit control over where tasks are spawned.

By default, a `Swarm` creates a `futures::executor::Threadpool` for these background tasks and that doesn't work for tokio. Also see #3068.